share/spack/gitlab/cloud_pipelines/stacks/ml/spack.yaml

```
@@ -0,0 +1,90 @@
+spack:
+  view: false
+
+  concretizer:
+    reuse: false
+    unify: false
+
+  config:
+    concretizer: clingo
+    install_tree:
+      root: /home/software/spack
+      padded_length: 384
+      projections:
+        all: '{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}'
+
```

If you don't specify an architecture in the environment, it will generate a pipeline for whatever specific arch was used in the `*-generate` job. Usually this causes problems when running the job so set it to something generic:
```suggestion
  packages:
    all:
      target: [x86_64_v4]
```